### PR TITLE
Skip every test suite if MIM is not running

### PR DESCRIPTION
**Pull request by arcusfelis**
This PR addresses MIM-2329.

Proposed changes include:
* Put in in the end of suite() callback
* Use require_rpc_nodes/2 instead of require_rpc_nodes/1

* Fail if MongoosIM is not running on required nodes. We have Common Test Hook doing it. Use `--skip-check-rpc-nodes` to disable.
* Fail if making an RPC to a non-required node. This is useful to ensure that the list or RPC nodes requested in the `suite()` callback is correct. Use `--allow-any-rpc` to disable the check.

* Added `cets_long:run_tracked` in a couple of places. `run_tracked` accepts a function and spawns a sidecar process, which would track the function execution. Each five seconds in will print time and current stacktrace.  Useful to debug MIM, which is stuck in starting.

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/244906) / elasticsearch_and_cassandra_mnesia / d761b5dd9faab3812bfc2d3210e1f64035823fff
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/244906/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/244906/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%408068351d119b.2024-12-03_14.43.20/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[dynamic_domains_mssql_mnesia_27](https://circleci.com/gh/esl/MongooseIM/245244) / odbc_mssql_mnesia / 18a8c642a5ed46c41bdddd30bfc6fdd9100751ae
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245244/odbc_mssql_mnesia.27.1.2-amd64)

**Comment by mongoose-im** [comment]
[pgsql_mnesia_27](https://circleci.com/gh/esl/MongooseIM/245267) / pgsql_mnesia / d50228741ed4f7e6a746f4fd3edea0a6e2e76b73
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245267/pgsql_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/index.html)
**OK: 5036** / **Failed: 3** / **User-skipped: 128** / **Auto-skipped: 109** 
<details><summary>component_SUITE:distributed:init_per_group</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.component_SUITE.logs/run.2024-12-03_19.39.24/component_suite.init_per_group.19074.html)
</details>
<details><summary>graphql_server_SUITE:init_per_suite</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.graphql_server_SUITE.logs/run.2024-12-03_19.44.26/graphql_server_suite.init_per_suite.html)
</details>
<details><summary>jingle_SUITE:init_per_suite</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.jingle_SUITE.logs/run.2024-12-03_19.44.52/jingle_suite.init_per_suite.html)
</details>
<details><summary>metrics_api_SUITE:all_metrics_are_global:init_per_group</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.metrics_api_SUITE.logs/run.2024-12-03_19.47.27/metrics_api_suite.init_per_group.212482.html)
</details>
<details><summary>metrics_api_SUITE:global:cluster_size</summary>


```erlang
{skip,
  {failed,
    {metrics_api_SUITE,init_per_testcase,
      {{test_case_failed,
         {assert_allowed_node,mongooseim2@localhost,
           [mongooseim@localhost]}},
       [{distributed_helper,rpc,4,
          [{file,
             "/home/circleci/project/big_tests/../test/common/distributed_helper.erl"},
           {line,136}]},
        {distributed_helper,has_mnesia,1,
          [{file,
             "/home/circleci/project/big_tests/../test/common/distributed_helper.erl"},
           {line,31}]},
        {distributed_helper,remove_node_from_cluster,2,
          [{file,
             "/home/circleci/project/big_tests/../test/common/distributed_helper.erl"},
           {line,53}]},
        {metrics_api_SUITE,'-ensure_nodes_not_clustered/1-lc$^1/1-1-',
          2,
          [{file,
             "/home/circleci/project/big_tests/tests/metrics_api_SUITE.erl"},
           {line,475}]},
        {metrics_api_SUITE,ensure_nodes_not_clustered,1,
          [{file,
             "/home/circleci/project/big_tests/tests/metrics_api_SUITE.erl"},
           {line,475}]},
        {metrics_api_SUITE,init_per_testcase,2,
          [{file,
             "/home/circleci/project/big_tests/tests/metrics_api_SUITE.erl"},
           {line,82}]},
        {test_server,do_init_per_testcase,2,
          [{file,"test_server.erl"},{line,1564}]}]}}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.metrics_api_SUITE.logs/run.2024-12-03_19.47.27/metrics_api_suite.cluster_size.html)
</details>
<details><summary>mod_global_distrib_SUITE:init_per_suite</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,reg1@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.mod_global_distrib_SUITE.logs/run.2024-12-03_19.48.02/mod_global_distrib_suite.init_per_suite.html)
</details>
<details><summary>persistent_cluster_id_SUITE:rdbms:all_nodes_in_the_cluster_have_the_same_cluster_id</summary>


```erlang
{skip,
  {failed,
    {persistent_cluster_id_SUITE,init_per_testcase,
      {{test_case_failed,
         {assert_allowed_node,mongooseim2@localhost,
           [mongooseim@localhost]}},
       [{distributed_helper,rpc,4,
          [{file,
             "/home/circleci/project/big_tests/../test/common/distributed_helper.erl"},
           {line,136}]},
        {distributed_helper,has_mnesia,1,
          [{file,
             "/home/circleci/project/big_tests/../test/common/distributed_helper.erl"},
           {line,31}]},
        {distributed_helper,add_node_to_cluster,2,
          [{file,
             "/home/circleci/project/big_tests/../test/common/distributed_helper.erl"},
           {line,34}]},
        {persistent_cluster_id_SUITE,init_per_testcase,2,
          [{file,
             "/home/circleci/project/big_tests/tests/persistent_cluster_id_SUITE.erl"},
           {line,102}]},
        {test_server,do_init_per_testcase,2,
          [{file,"test_server.erl"},{line,1564}]},
        {test_server,run_test_case_eval1,6,
          [{file,"test_server.erl"},{line,1265}]},
        {test_server,run_test_case_eval,9,
          [{file,"test_server.erl"},{line,1235}]}]}}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.persistent_cluster_id_SUITE.logs/run.2024-12-03_19.50.23/persistent_cluster_id_suite.all_nodes_in_the_cluster_have_the_same_cluster_id.html)
</details>
<details><summary>persistent_cluster_id_SUITE:rdbms:clean_start_and_two_nodes</summary>


```erlang
{error,
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.persistent_cluster_id_SUITE.logs/run.2024-12-03_19.50.23/persistent_cluster_id_suite.clean_start_and_two_nodes.html)
</details>
<details><summary>pubsub_s2s_SUITE:init_per_suite</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,fed1@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.pubsub_s2s_SUITE.logs/run.2024-12-03_19.50.51/pubsub_s2s_suite.init_per_suite.html)
</details>
<details><summary>rest_client_SUITE:security:non_default_http_server_name_is_returned_if_configured</summary>


```erlang
{error,
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.rest_client_SUITE.logs/run.2024-12-03_19.52.18/rest_client_suite.non_default_http_server_name_is_returned_if_configured.html)
</details>
<details><summary>service_mongoose_system_metrics_SUITE:init_per_suite</summary>


```erlang
{'EXIT',
  {test_case_failed,
    {assert_allowed_node,mongooseim2@localhost,[mongooseim@localhost]}}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.service_mongoose_system_metrics_SUITE.logs/run.2024-12-03_19.53.42/service_mongoose_system_metrics_suite.init_per_suite.html)
</details>
<details><summary>system_probes_SUITE:mnesia:mnesia_info</summary>


```erlang
{error,{test_case_failed,"Incorrect number of instrumentation events - matched: 0, expected: positive"}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.system_probes_SUITE.logs/run.2024-12-03_19.54.28/system_probes_suite.mnesia_info.html)
</details>
<details><summary>system_probes_SUITE:end_per_suite</summary>


```erlang
{error,{test_case_failed,"Instrumentation events that were logged, but not tested:\n[{mnesia_info,#{}}]\nYou need to test them with instrument_helper:assert/3"}}
```

[Report log](https://esl.github.io/html-zip-reader/PR/4413/245267/pgsql_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4040a9a397d330.2024-12-03_19.37.44/big_tests.tests.system_probes_SUITE.logs/run.2024-12-03_19.54.28/system_probes_suite.end_per_suite.html)
</details>

**Comment by mongoose-im** [comment]
[dynamic_domains_pgsql_mnesia_26](https://circleci.com/gh/esl/MongooseIM/245299) / pgsql_mnesia / 617e858ac4bb843e5a76d848babca2a0caace76e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245299/pgsql_mnesia.26.2.5.4-amd64)

**Comment by mongoose-im** [comment]
[dynamic_domains_pgsql_mnesia_26](https://circleci.com/gh/esl/MongooseIM/245352) / pgsql_mnesia / 6a7eb6d57a8869e4873d89afc32c4df97622dbc1
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245352/pgsql_mnesia.26.2.5.4-amd64)

**Comment by mongoose-im** [comment]
[dynamic_domains_pgsql_mnesia_26](https://circleci.com/gh/esl/MongooseIM/245405) / pgsql_mnesia / 30de7d379cb231a074ff674274df9a9defb464ea
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245405/pgsql_mnesia.26.2.5.4-amd64)

**Comment by mongoose-im** [comment]
[dynamic_domains_pgsql_mnesia_26](https://circleci.com/gh/esl/MongooseIM/245442) / pgsql_mnesia / 784b94ad03ef8789608d7be252ec7cb2bbbd0f62
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245442/pgsql_mnesia.26.2.5.4-amd64)

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245488) / elasticsearch_and_cassandra_mnesia / d862adfd931ec812e00df263bc24a9d8bd137235
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245488/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245488/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%404dc7a046b1db.2024-12-03_23.52.02/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245528) / elasticsearch_and_cassandra_mnesia / 4d505a7f79a41efe060253850267a004b15fe5c6
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245528/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245528/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40818c7eb4371f.2024-12-04_00.01.19/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245576) / elasticsearch_and_cassandra_mnesia / 4e64c534908b305900c35195e5c37cd3bd079727
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245576/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245576/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%409ba783fe3383.2024-12-04_01.22.08/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245620) / elasticsearch_and_cassandra_mnesia / 92ac66ea1aab7f1bb30c58abb0879a10f08bdf68
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245620/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245620/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40be01dfc545ad.2024-12-04_01.50.39/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245665) / elasticsearch_and_cassandra_mnesia / dbdec89ae740718b59d8aa4b05907ab9200afc15
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245665/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245665/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40a78a279d3265.2024-12-04_02.02.11/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245703) / elasticsearch_and_cassandra_mnesia / 030160e4fd389a8f29548d5476574e95aba90429
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245703/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245703/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%405e222bd40121.2024-12-04_02.29.44/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245748) / elasticsearch_and_cassandra_mnesia / e78e3a1aa91b00ffaf2aacb9968483534d88f69b
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245748/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245748/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%404dd04ccf032f.2024-12-04_08.26.30/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245794) / elasticsearch_and_cassandra_mnesia / 27ec2dd026e336759e99492ba6986786ef447bdf
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245794/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245794/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40c8182c45cda6.2024-12-04_08.51.04/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245859) / elasticsearch_and_cassandra_mnesia / 69385b7a702d7d7922cf6d49e2cee2fa7d3cc403
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245859/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245859/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40fd744ff509e1.2024-12-04_10.13.32/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/245997) / elasticsearch_and_cassandra_mnesia / 16595e1af01a47ec42173fd3c044731e0ab2112c
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/245997/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/245997/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40ab9a72e967a2.2024-12-04_11.02.33/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/246037) / elasticsearch_and_cassandra_mnesia / fe274a8ff506d02f7731ad6770bb079717f7121d
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/246037/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/246037/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40ef7562eca227.2024-12-04_11.34.59/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/246133) / elasticsearch_and_cassandra_mnesia / 6189c95a83805815051005c3e3355c5d031a931a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/246133/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/246133/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4059d886da2779.2024-12-04_12.18.19/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/246172) / elasticsearch_and_cassandra_mnesia / 69a00e104601cc53a1bdfe2e6476a44cdb65c0fa
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/246172/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/246172/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40365d7b17784b.2024-12-04_12.25.56/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/246240) / elasticsearch_and_cassandra_mnesia / 0a25252f01126ad5cc77d6be91e8117815043132
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/246240/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/246240/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%401a1d698bfdd5.2024-12-04_13.46.50/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: tools/test-runner.sh [hunk 1]
```diff
@@ -115,6 +117,9 @@ Script examples:
     Sets dev-nodes and test-hosts to empty lists
     Reruns mam_SUITE
 
+./tools/test-runner.sh --skip-small-tests --skip-check-rpc-nodes --allow-any-rpc --one-node connect
+    Allows any RPCs in the tests, useful if not all nodes are running on the developer machine
+
```
Comment (NelsonVides) [inline tools/test-runner.sh:122]: The `--one-node` flag, I use it very often, does this mean that now I have to use both together? Under what scenario would I want `--one-node` without `--allow-any-rpc`?

File: big_tests/tests/inbox_extensions_SUITE.erl [hunk 2]
```diff
@@ -25,7 +25,10 @@
          end_per_testcase/2]).
 
 all() ->
-    inbox_helper:skip_or_run_inbox_tests(tests()).
+    Tests = tests(),
+    distributed_helper:temporary_allow_nodes(fun() ->
+            inbox_helper:skip_or_run_inbox_tests(Tests)
+        end, [mim]).
```
Comment (NelsonVides) [inline big_tests/tests/inbox_extensions_SUITE.erl:31]: I'm not sure to understand when am I supposed to use `temporary_allow_nodes/1` 🤔

File: tools/test-runner.sh (hunk 1 above)
Comment (arcusfelis) [reply tools/test-runner.sh:122]: ok, I added automatically assumed `--skip-check-rpc-nodes` and `--allow-any-rpc` to `--one-node`.
The reasoning for this new change is:
- if we use test-runner for running just one test, which we often do, we don't need RPC nodes checks thaat much.


> Under what scenario would I want --one-node without --allow-any-rpc?
When you have suite which uses only one node.
Still, sometimes we have suites which use more than one node, but we ok with tests to fail during dev. So, made different behaviour for `--one-node`.


Theoretically, we can extract `TEST_HOSTS` in the code and only check nodes which are provided in `TEST_HOSTS` against required list. And we would only control `TEST_HOSTS` instead.... mmm, maybe it is a good idea actually :D

File: big_tests/tests/inbox_extensions_SUITE.erl (hunk 2 above)
Comment (arcusfelis) [reply big_tests/tests/inbox_extensions_SUITE.erl:31]: in code which is called from `groups/0` CT callback. Or from `all/0` callback.
I.e. almost never. 

Unless you call something like an RPC to a node to figure out which tests to run... like here.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/246729) / elasticsearch_and_cassandra_mnesia / f111277d89f2a51c0d99f901cd8bd471d9b2053e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/246729/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/246729/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%409489249b80d8.2024-12-05_10.42.31/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/246765) / elasticsearch_and_cassandra_mnesia / c3414df3b939d50c4c1513d573de16b52a76e918
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/246765/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/246765/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%400d7bc0f3f186.2024-12-05_11.13.15/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: big_tests/tests/inbox_extensions_SUITE.erl (hunk 2 above)
Comment (arcusfelis) [reply big_tests/tests/inbox_extensions_SUITE.erl:31]: Technically, this part of code smell a bit.
Maybe we need a helper module, which would have something like:
```
preset_info() -> #{rdbms => true, elasticsearch => true, is_mssql => false}.
```
and so on.

And we can actually decide if we wanna do an RPC or put these `test.config` presets.

By the way, we don't need to do RPC at all:
```
    {odbc_mssql_mnesia,
     [{dbs, [redis, mssql]},
```

`test.config` has a list of databases, that are supposed to be running during a preset.

Could refactor calls in `all/0` and `groups/0`. But sounds like a separate PR.
So, after that `temporary_allow_nodes` workaround could be removed at all.

 + during that task we can check that testname autocomplition works again (because if you do RPCs, it does not. And that is why there is also `is_ct_running` call usually, but for example, it is missing here too).
 
 Just grabbing DB type from the running node is the most "obvious" solution usually.

Comment (arcusfelis) [reply big_tests/tests/inbox_extensions_SUITE.erl:31]: Also, funny, we get preset inside `Config`. But not in `groups/0` or `all/0` callbacks. 
So, we would have to pass preset as OS env variable probably.

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/247311) / elasticsearch_and_cassandra_mnesia / 34ebed08117e28d4598ddf6ff35c8bd94291a73a
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/247311/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/247311/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%40d55c8a3e1d44.2024-12-05_15.59.27/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/247339) / elasticsearch_and_cassandra_mnesia / 2972994d208c897271f72e0eda36b23a7dddfe68
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4413/247339/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4413/247339/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%4042b1d43967fb.2024-12-05_16.35.16/index.html)
**OK: 473** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

File: src/mongoose_internal_databases.erl [hunk 3]
```diff
@@ -28,11 +30,32 @@ init(mnesia, #{}) ->
...
     application:start(mnesia, permanent),
-    mnesia:wait_for_tables(mnesia:system_info(local_tables), infinity),
+    wait_for_tables_loop(mnesia:system_info(local_tables), 10000, 0),
     mongoose_node_num_mnesia:init();
 init(cets, #{}) ->
     ok.
 
+%% Sometimes mnesia:wait_for_tables/2 could hang on startup.
+%% This function logs which tables are not ready and their status.
+wait_for_tables_loop(Tables, Interval, Total) ->
```
Comment (chrzaszcz) [inline src/mongoose_internal_databases.erl:40]: This seems unrelated. If there is a new issue, it should be logged and resolved separately IMO.

File: src/mongoose_cluster.erl [hunk 4]
```diff
@@ -212,11 +214,12 @@ detach_nodes(Nodes) ->
...
     mnesia:system_info(db_nodes) -- [node()].
 
-with_app_stopped(App, F) ->
+with_app_stopped(App, F, Info) ->
     Running = is_app_running(App),
     Running andalso application:stop(App),
     try
-        F()
+        %% Reports the progress if it takes too long
+        cets_long:run_tracked(Info, F)
```
Comment (chrzaszcz) [inline src/mongoose_cluster.erl:222]: I wouldn't use CETS code in mnesia-related modules. It creates confusion.

File: src/ejabberd_sup.erl [hunk 5]
```diff
@@ -121,7 +124,17 @@ worker_spec(Mod) ->
...
+    %% in the slow init worker functions.
+    MFA = {?MODULE, start_linked_child, [Mod, Args]},
+    {Mod, MFA, permanent, timer:seconds(5), worker, [Mod]}.
+
+%% In case one of the workers takes long time to start
+%% we want the logging progress (to know which child got stuck).
+%% This could happend on CI during the node restarts.
+start_linked_child(Mod, Args) ->
+    F = fun() -> erlang:apply(Mod, start_link, Args) end,
+    cets_long:run_tracked(#{task => start_linked_child, child_module => Mod}, F).
```
Comment (chrzaszcz) [inline src/ejabberd_sup.erl:137]: This doesn't seem to be CETS-related code... I am not liking these ad-hoc dependencies between MongooseIM and CETS.

**Review by chrzaszcz — COMMENTED** [review]
Looks good in general. I haven't checked the suites one by one, but I have a few general comments:
- Could we just do without limiting the RPC calls to required nodes? It would make the code easier to understand, and less complicated.
- I think that the source code changes don't belong to this PR, i.e. don't modify tests and code at once, unless required.

**Comment by arcusfelis** [comment]
> Could we just do without limiting the RPC calls to required nodes? It would make the code easier to understand, and less complicated.
We had it before. And we basically had incorrect require_rpc_nodes in 50% of suites.

**Comment by arcusfelis** [comment]
closed because https://github.com/esl/MongooseIM/pull/4426/files is shorter
